--- a/drivers/thermal/qcom/tsens-v0_1.c
+++ b/drivers/thermal/qcom/tsens-v0_1.c
@@ -83,20 +83,25 @@
 	},
 };
 
 static int calibrate_8916(struct tsens_priv *priv)
 {
 	u32 p1[5], p2[5];
 	u32 *qfprom_cdata, *qfprom_csel;
 	int mode, ret;
 
 	ret = tsens_calibrate_nvmem(priv, 3);
+
+	if (ret == -EPROBE_DEFER) {
+		return ret;
+	}
+
 	if (!ret)
 		return 0;
 
 	qfprom_cdata = (u32 *)qfprom_read(priv->dev, "calib");
 	if (IS_ERR(qfprom_cdata))
 		return PTR_ERR(qfprom_cdata);
 
 	qfprom_csel = (u32 *)qfprom_read(priv->dev, "calib_sel");
 	if (IS_ERR(qfprom_csel)) {
 		kfree(qfprom_cdata);
